File: easy/The Descent/Go/The Descent.go
package main

import "fmt"

/**
 * The while loop represents the game.
 * Each iteration represents a turn of the game
 * where you are given inputs (the heights of the mountains)
 * and where you have to print an output (the index of the mountain to fire on)
 * The inputs you are given are automatically updated according to your last actions.
 **/

func main() {
    for {
        biggest := 0
        num := 0

        for i := 0; i < 8; i++ {
            // mountainH: represents the height of one mountain.
            var mountainH int
            fmt.Scan(&mountainH)

            if mountainH > biggest {
                biggest = mountainH
                num = i
            }
        }
        
        // fmt.Fprintln(os.Stderr, "Debug messages...")
        fmt.Println(num) // The index of the mountain to fire on.
    }
}

File: easy/The Descent/Perl/The Descent.pl
use strict;
use warnings;
#use diagnostics;
use 5.32.1;

select(STDOUT); $| = 1; # DO NOT REMOVE

# The while loop represents the game.
# Each iteration represents a turn of the game
# where you are given inputs (the heights of the mountains)
# and where you have to print an output (the index of the mountain to fire on)
# The inputs you are given are automatically updated according to your last actions.


# game loop
while (1) {
    my $biggest = 0;
    my $num = 0;

    for my $i (0..7) {
        chomp(my $mountain_h = <STDIN>); # represents the height of one mountain.
        if ($biggest < $mountain_h) {
            $biggest = $mountain_h;
            $num = $i
        }
    }
    
    # Write an action using print
    # To debug: print STDERR "Debug messages...\n";

    print "$num\n"; # The index of the mountain to fire on.
}